zrp/prepare/geo_geocoder.py

```diff
@@ -139,34 +139,21 @@ def transform(self, input_data, geo, processed, replicate, save_table=True):
         data["ZEST_FULLNAME"] = data[self.street_address]
         data['ZEST_KEY_LONG'] = data[[self.key, 'replicate_flg']].apply(lambda x: "".join(x.dropna()), axis=1)
         print("      ...merge user input & lookup table")
-        print(data.shape)
         geo_df = data.merge(aef, on=["ZEST_FULLNAME"], how="left")
 
-        ########
-        #Here non-numeric HN matching could be added
-        geo_df['FROMHN_numeric'] = geo_df.FROMHN.apply(lambda x: int(x) if str(x).isnumeric() else -1)
-        geo_df['TOHN_numeric'] = geo_df.TOHN.apply(lambda x: int(x) if str(x).isnumeric() else -1)
-        geo_df["house_numer_numeric"] = geo_df[self.house_number].apply(lambda x: int(x) if str(x).isnumeric() else -1)       
-        ########
-        
-        geo_df["small"] = np.where(
-            (geo_df.FROMHN_numeric > geo_df.TOHN_numeric),
-            geo_df.TOHN_numeric,
-            geo_df.FROMHN_numeric)
-
-        geo_df["big"] = np.where(
-            (geo_df.FROMHN_numeric > geo_df.TOHN_numeric),
-            geo_df.FROMHN_numeric,
-            geo_df.TOHN_numeric)
-      
+        geo_df['FROMHN_RIGHT'] = geo_df['FROMHN_RIGHT'].fillna(-2).astype(float).astype(int)
+        geo_df['TOHN_RIGHT'] = geo_df['TOHN_RIGHT'].fillna(-2).astype(float).astype(int)
+        geo_df["house_number_RIGHT"] = geo_df[self.house_number+'_RIGHT'].replace("", np.nan).fillna(-1).astype(float).astype(int)
```

Comment:
Why fillna? Does this potentially create some edge cases where boundaries are inverted?

Reply:
We needed to get rid of na’s to be able to convert numbers to ints – fillna had better performance then appropriate apply function. Since FROHMN_RIGHT/LEFT are taken from the lookup tables and there are no negative values there the solution doesn’t rise any issues.